Source/gg2/Objects/Weapons/Medigun.events/User Event 1.xml

```diff
@@ -16,12 +16,32 @@
         <argument kind="STRING">{
     if(!instance_exists(healTarget) and readyToShoot)
     {
-        var bestTarget, bestTargetDist;
+        var bestTarget, bestTargetDist, testx, testy;
         bestTarget=noone;
-        bestTargetDist=maxHealDistance;
+        bestTargetDist = maxHealDistance;
+        
+        testx = round(owner.x) + cos(degtorad(owner.aimDirection))*owner.aimDistance;
+        testy = round(owner.y) - sin(degtorad(owner.aimDirection))*owner.aimDistance;
 
         with(Character) {
-            if(id != other.owner and team == other.owner.team) {
+            if point_distance(x,y,other.owner.x,other.owner.y) &lt;= other.maxHealDistance
+               &amp;&amp; point_distance(x,y,testx,testy) &lt; bestTargetDist {
+                if(id != other.owner and (team == other.owner.team)) {
+                    if(collision_line(other.x, other.y,
+                    other.x+lengthdir_x(other.maxHealDistance, other.owner.aimDirection),
+                    other.y+lengthdir_y(other.maxHealDistance, other.owner.aimDirection), id, true, false))
+                    {
+                        if(!collision_line(other.x, other.y, x, y, Obstacle, true, true)) {
+                            bestTargetDist = point_distance(x,y,testx,testy);
+                            bestTarget = player;
+                        }
+                    }
+                }
+            }
+        }
+        
+        if bestTarget = noone with(Character) {
```

Comment:
I used maxHealDistance because I was lazy and didn't feel like defining another variable. It's arbitrary and I agree that it should be changed. I felt that making the distance too short would have been worse than making it too long, so I overshot with a figure familiar to most people. It gives me a simple way to explain it to them when they ask.

I'll go back over it in a little while. And I'll get some folks' opinions on what a good distance would be. And I'll deduplicate the code.